### PR TITLE
Enrich Command

### PR DESCRIPTION
**Pull request by Alexander-Baker-24**
Finished writing the enrich command and updated mkindex to commit 68d90b4.

### CODE REVIEW

File: cblite/EnrichCommand.cc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            {
+                C4Warn("REST request failed. %d %d", r->error().domain, r->error().code);
+                r->error() = C4Error{NetworkDomain, kC4NetErrUnknown};
+            }
+            return;
+        }
+        
+        // Parse response
+        Doc newDoc = Doc::fromJSON(response);
+        Value val = newDoc.asDict().get("data").asArray()[0].asDict().get("embedding");
```
Comment (borrrden) [inline cblite/EnrichCommand.cc:136]: Let's choose a better name than `val` as well.  `embedding` seems like a natural choice.

File: cblite/EnrichCommand.hh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+class EnrichCommand : public CBLiteCommand {
+public:
+    EnrichCommand(CBLiteTool &parent)
+    :CBLiteCommand(parent)
+    { }
+    void usage() override;
+    void runSubcommand() override;
+protected:
+    void enrichDocs(std::string, std::string);
```
Comment (borrrden) [inline cblite/EnrichCommand.hh:31]: I think this could probably be `const std::string&` instead

File: tests/TokenizerTest.cc [hunk 3]
```diff
@@ -1,4 +1,4 @@
-//
+    //
```
Comment (borrrden) [inline tests/TokenizerTest.cc:1]: Seems like this was a mistake?

File: vendor/couchbase-lite-core-EE
Comment (borrrden) [inline vendor/couchbase-lite-core-EE:1]: This cannot be committed, it needs to be removed

**Review by borrrden — CHANGES_REQUESTED** [review]
The PR also only mentions the enrich command, with no mention of the modifications to the mkindex command.

File: cblite/EnrichCommand.cc [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        if (r->run()) {
+            response = r->body();
+        } else {
+            if ( r->error() == C4Error{NetworkDomain, kC4NetErrTimeout} ) {
+                C4Warn("REST request timed out. Current timeout is %f seconds", r->getTimeout());
+            }
+            else
+            {
+                C4Warn("REST request failed. %d %d", r->error().domain, r->error().code);
+                r->error() = C4Error{NetworkDomain, kC4NetErrUnknown};
```
[the remark this answers is not in the document]
Comment (borrrden) [reply cblite/EnrichCommand.cc:127]: Same question

File: tests/TokenizerTest.cc (hunk 3 above)
Comment (borrrden) [reply tests/TokenizerTest.cc:1]: This didn't get fixed?

File: cblite/EnrichCommand.hh (hunk 2 above)
Comment (borrrden) [reply cblite/EnrichCommand.hh:31]: Why was this conversation resolved if nothing was changed?

File: cblite/EnrichCommand.cc [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        // Convert body to json
+        slice propName = property.asString();
+        cout << propName.asString() << "\n";
+        auto mutableBody = body.mutableCopy(kFLDefaultCopy);
+        mutableBody.set(dstProp, property);
+        auto json = mutableBody.toJSON();
+          
+        alloc_slice newBody = alloc_slice(c4db_encodeJSON(_db, json, &error));
+        if (!newBody)
+            fail("Couldn't encode body", error);
```
[the remark this answers is not in the document]
Comment (borrrden) [reply cblite/EnrichCommand.cc:99]: There is no need anymore to convert to body to JSON.  This was not addressed.

File: Tool.hh [hunk 6]
```diff
@@ -255,7 +260,15 @@ protected:
...
-            if (!processFlag(flag, specs)) {
+            
+                bool handled;
+                try {
+                    handled = processFlag(flag, specs);
+                } catch (std::exception const& x) {
+                    fail("in flag " + flag + ": " + x.what());
+                }
+
+                if (!handled) {
```
Comment (borrrden) [inline Tool.hh:271]: This entire block is incorrectly indented

**Review by borrrden — CHANGES_REQUESTED** [review]

File: cblite/EnrichCommand.cc (hunk 4 above)
[the remark this answers is not in the document]
Comment (Alexander-Baker-24) [reply cblite/EnrichCommand.cc:127]: I believe it warns about an error. I can delete it if it is not needed.

[the remark this answers is not in the document]
Comment (borrrden) [reply cblite/EnrichCommand.cc:117]: I meant the line after that.  `r->error() = C4Error{NetworkDomain, kC4NetErrUnknown};`

File: cblite/EnrichCommand.cc [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    "    " << it("DESTINATION") << " : destination property\n"
+    ;
+}
+
+void EnrichCommand::runSubcommand() {
+    // Read params:
+    processFlags({
+        {"--offset", [&]{offsetFlag();}},
+        {"--limit",  [&]{limitFlag();}},
+        //{"--writeable", [&]{writeableFlag();}}
```
Comment (borrrden) [inline cblite/EnrichCommand.cc:48]: If this comment is not needed then remove it

**Review by borrrden — CHANGES_REQUESTED** [review]
Just one last little thing I found

**Review by borrrden — APPROVED** [review]

**Pull request merged**